HighwayPatrol.cpp

```diff
@@ -1,11 +1,16 @@
 #include "HighwayPatrol.h"
+
 #include <iostream>
 
 HighwayPatrol::HighwayPatrol() : Vehicle("HighwayPatrol")
 {
 
 }
 
+HighwayPatrol::~HighwayPatrol() = default;
+HighwayPatrol::HighwayPatrol(const HighwayPatrol&) = default;
+HighwayPatrol& HighwayPatrol::operator=(const HighwayPatrol&) = default;
+
 void HighwayPatrol::scanHighway(Highway* h)
 {
     std::cout << name << ": scanning highway for speeders" << std::endl;
```

Comment:
the `for()` loop below should be: 
```
for( size_t i = h->vehicles.size(); --i != 0; )
{
    auto* v = h->vehicles[i];
```
this was a bug caught by another user.  `pullOver` will remove the vehicle from the `vehicles` vector, causing undefined behavior if `vehicle` was in the middle of the list of `vehicles`.
- [ ] please revise.